### PR TITLE
Kube Rebase 1.29

### PR DESCRIPTION
**Pull request by bpickard22**
Update K8s deps in in ovnk and in test dir to 1.29

Lots of small build fixes regarding klog no longer supporting wrapped errors and a few cases of incorrect number of variables referenced in an error or info return

### CODE REVIEW

**Comment by tssurya** [comment]
please fix CI

**Comment by tssurya** [comment]
CI is still red here putting hold on this, please mark this ready for reviews when entire CI is green, I'd like to also get the IPV6 lanes merged before merging this one: https://github.com/ovn-org/ovn-kubernetes/pull/4106

**Comment by tssurya** [comment]
upgrade and the 3 shard conformance lanes need some love

File: test/conformance/vendor/github.com/davecgh/go-spew/LICENSE
Comment (tssurya) [inline test/conformance/vendor/github.com/davecgh/go-spew/LICENSE:1]: wait why are we pulling in all this, this directory is light weight maintenance like test/e2e and we don't vendor in stuff here.

**Review by tssurya — CHANGES_REQUESTED** [review]
hey I can't even load this PR, my browser page hangs, maybe I need t

File: test/conformance/vendor/github.com/davecgh/go-spew/LICENSE
Comment (bpickard22) [reply test/conformance/vendor/github.com/davecgh/go-spew/LICENSE:1]: probably pulled in when i updated the k8s versions in the go mod in the conformance dir and ran tidy and vendor, is that an issue?

Comment (bpickard22) [reply test/conformance/vendor/github.com/davecgh/go-spew/LICENSE:1]: nvm found the issue in the e2e ci and fixed that

**Comment by bpickard22** [comment]
still need to fix the issue with the serviceCidr feature but this adds the kind and test dir fixes on top of @tssurya pr

**Comment by bpickard22** [comment]
resolved conflicts, working on the build errors from kind currently

File: contrib/kind.yaml.j2
Comment (trozet) [inline contrib/kind.yaml.j2]: what caused you to add this?

Comment (bpickard22) [reply contrib/kind.yaml.j2]: my intial plan was to enable the serviceCidr feature gate in kind support the feature but it caused lots of issues with dualstack and live migration so I decided to skip the serviceCidr test and say we dont support it for now (its a v1alpha feature) 
I will open an issue noting this

Comment (tssurya) [reply contrib/kind.yaml.j2]: yeah eventually when it does graduate to beta we must add support for it. hmm too bad I thought it would have been straightforward.

**Comment by bpickard22** [comment]
@tssurya yes this is ready for review

File: contrib/kind.yaml.j2
Comment (tssurya) [reply contrib/kind.yaml.j2]: @bpickard22 : qq re https://github.com/ovn-org/ovn-kubernetes/issues/4269
If the feature is enabled by default aren't we supposed to turn off the feature gate? why is this set to true? or does this mean the gate is ON, which means the feature is off?

File: test/conformance/go.mod
Comment (tssurya) [inline test/conformance/go.mod:5]: why is there a toolchain here?

I think its because you have not updated net-pol-api itself I think you can drop this commit from this PR and I'll get it in via https://github.com/ovn-org/ovn-kubernetes/pull/4245/commits/c918793782dc87886dad1fe2fa404544ce1b5ac2 ? wdyt?

File: contrib/kind.yaml.j2
Comment (tssurya) [inline contrib/kind.yaml.j2]: hmm I am confused a little as to why this was added in first commit and is now deleted in 4th commit?

File: test/e2e/e2e.go
Comment (tssurya) [inline test/e2e/e2e.go:1]: I am not a fan of vendoring in the e2e's is there a reason why we are having to do this?

File: test/e2e/go.mod
Comment (tssurya) [inline test/e2e/go.mod:5]: why do we have toolchain coming here again? is your local version not aligned with what we need to 1.29.2?

File: test/e2e/external_gateways.go
Comment (tssurya) [inline test/e2e/external_gateways.go:3008]: is framework.ExceptEqual just gone? commit didn't pint to relevant changes in k/k where that happened.

**Review by tssurya — COMMENTED** [review]
lgtm on the main logic, some comments inline before I approve (ofc tests have to pass :D) if you see a UT flake please tag the appropriate issue with ci-flake if one exists upstream, if not open a new one if its a new flake.

File: contrib/kind.yaml.j2
Comment (bpickard22) [reply contrib/kind.yaml.j2]: it is not enabled by default, this was part of my attempt to enable it hoping that what needed to be plumbed would happen and everything would be fine, which was not the case, i just didnt drop that change from the commit and instead removed it when i skipped the test

File: test/conformance/go.mod
Comment (bpickard22) [reply test/conformance/go.mod:5]: yeah i agree, will drop the commit

File: contrib/kind.yaml.j2
Comment (bpickard22) [reply contrib/kind.yaml.j2]: did a bad job of cleaning my commits up, sorry for the confusion

File: test/e2e/go.mod
Comment (bpickard22) [reply test/e2e/go.mod:5]: no my local go is on 1.21, im not sure why this is here, will remove it

File: contrib/kind.yaml.j2
Comment (bpickard22) [reply contrib/kind.yaml.j2]: fixed

File: test/e2e/e2e.go
Comment (bpickard22) [reply test/e2e/e2e.go:1]: no i can drop the /vendor changes

File: contrib/kind.yaml.j2
Comment (bpickard22) [reply contrib/kind.yaml.j2]: > @bpickard22 : qq re https://github.com/ovn-org/ovn-kubernetes/issues/4269
If the feature is enabled by default aren't we supposed to turn off the feature gate? why is this set to true? or does this mean the gate is ON, which means the feature is off?

to answer this, not turning on the feature gate turn the feature on, but the issue is that the k8s e2e for this is running regardless of if the feature is on or not, and we dont currently support this in ovn-k. I opened an issue to track this -> https://github.com/ovn-org/ovn-kubernetes/issues/4269 

i know we had thought that turning on the feature gate would handle the this, but @martinkennelly and i explored this option and decided to skip the test for now because it broke dualstack and live migration as well as caused some issues with kubevirt, i will work on enabling it when i have more time if that is fine with you and @trozet

Comment (bpickard22) [reply contrib/kind.yaml.j2]: I think this explains it better than I did before

File: test/scripts/e2e-kind.sh
Comment (tssurya) [inline test/scripts/e2e-kind.sh:66]: nit: extra while lines

File: test/scripts/e2e-kind.sh
Comment (tssurya) [inline test/scripts/e2e-kind.sh:63]: is `servce` expected :P ? did they also type the tests wrongly upstream in k/k ? :D

**Comment by tssurya** [comment]
1. control-plane v6 flake is known: https://github.com/ovn-org/ovn-kubernetes/issues/4275
2. kv-live-migration is also known, infact has been perma failing for weeks now: https://github.com/ovn-org/ovn-kubernetes/issues/4237

**Review by tssurya — COMMENTED** [review]
last comment, besides that its lgtm
can you please double check your skip

File: test/scripts/e2e-kind.sh
Comment (bpickard22) [reply test/scripts/e2e-kind.sh:63]: yes that is the spelling upstream

Comment (tssurya) [reply test/scripts/e2e-kind.sh:63]: lol

**Comment by tssurya** [comment]
re running shard conformance, once that passes will merge this.

File: .github/workflows/test.yml
Comment (tssurya) [inline .github/workflows/test.yml:18]:
```suggestion
  GO_VERSION: "1.21.6"
```
? be consistent? pick either or , I see the previous file using one version and this one is doing another

File: test/e2e/go.mod
Comment (tssurya) [inline test/e2e/go.mod:5]: why? I think your local go version is not aligning, its better to use one verison 1.21, 1.21.6 1.21.8 I see its all over the place... pick the version which is actually being used by k/k 1.29.2 (go check which version is used in that release and use that same golang version)

**Review by tssurya — CHANGES_REQUESTED** [review]

File: test/e2e/go.mod
Comment (bpickard22) [reply test/e2e/go.mod:5]: yes my local go is 1.21.8, will correct all of these to be 1.21 which is what 1.29 k/k is on

File: .github/workflows/test.yml
Comment (bpickard22) [reply .github/workflows/test.yml:18]: k/k is on 1.21 so i will stick with this

File: .github/workflows/docker.yml
Comment (tssurya) [inline .github/workflows/docker.yml:12]:
```suggestion
  GO_VERSION: 1.21
```
also here?

**Review by tssurya — COMMENTED** [review]
4 lanes are red, please link them to known flakes so that its easy to know if its known or not or debug them, cannot merge PR with red CI (see sample of how I did it in the above comment for your previous run)

**Comment by bpickard22** [comment]
both red lanes failing on known control plane ipv6 issue
1: https://github.com/ovn-org/ovn-kubernetes/issues/4275

**Comment by tssurya** [comment]
CI is green, changes look good merging this. thanks @bpickard22 !

**Review by tssurya — APPROVED** [review]

**Pull request merged**
